src/components/validation/mod.rs

```diff
@@ -102,14 +104,9 @@ where
     fn external_resource(&self) -> Option<ExternalResource> {
         (*self).external_resource()
     }
-
-    fn test_cases(&self) -> Vec<TestCase> {
-        (*self).test_cases()
-    }
 }
 
-#[cfg(feature = "sources-http_server")]
-#[cfg(test)]
+#[cfg(all(test, feature = "validation-tests"))]
```

Comment:
:thinking:  wondering if this feature should be something like "component-validation-tests" , simply because all tests are presumably validating something :)

Reply:
Heh, yeah, that's fair. It's a bit ambiguous as-is.

I'll make that change.

Reply:
Took care of this in 15525b31a.